src/autopas/containers/linkedCells/LinkedCells.h

```diff
@@ -116,15 +116,11 @@ class LinkedCells : public ParticleContainer<Particle, ParticleCell, SoAArraysTy
     traversal->endTraversal(this->_cells);
   }
 
-  void updateContainer() override {
-    auto haloIter = this->begin(IteratorBehavior::haloOnly);
-    if (haloIter.isValid()) {
-      utils::ExceptionHandler::exception(
-          "Linked Cells: Halo particles still present when updateContainer was called. First particle found:\n" +
-          haloIter->toString());
-    }
-
+  std::vector<Particle> AUTOPAS_WARN_UNUSED_RESULT updateContainer() override {
+    this->deleteHaloParticles();
+    std::vector<Particle> invalidParticles;
 #ifdef AUTOPAS_OPENMP
+#pragma omp declare reduction(vecMerge : std::vector<Particle> : omp_out.insert(omp_out.end(), omp_in.begin(), omp_in.end()))
```

Comment:
Move to WrapOpenMP.h

Reply:
not possible, as it uses a template

Reply:
actually, this isn't possible at all. now using critical sections for this...